### PR TITLE
Make nvcc code generation target configure by env var

### PR DESCRIPTION
**Pull request by take-cheeze**
Like ChainerX: https://github.com/chainer/chainer/blob/c8e47f7e8463c84cedb6ac12ccbad582d6321ce5/chainerx_cc/CMakeLists.txt#L168-L170

### CODE REVIEW

File: cupy_setup_build.py [hunk 1]
```diff
@@ -637,6 +637,13 @@ def get_ext_modules(use_cython=False):
 def _nvcc_gencode_options(cuda_version):
     """Returns NVCC GPU code generation options."""
 
+    if sys.argv == ['setup.py', 'develop']:
+        return []
+
+    envcfg = os.getenv('CUPY_NVCC_GENERATE_CODE', None)
+    if envcfg is not None:
```
Comment (okapies) [inline cupy_setup_build.py]: In ChainerX, the build configuration skips adding the argument if `envcfg` is empty string. I have no idea which behavior is preferred (`""` will override the default behavior and add just a `--generate-code=`). Any idea?

Comment (take-cheeze) [reply cupy_setup_build.py]: Thanks for mentioning.
I think empty string should be ignore too will fix this.

**Comment by okapies** [comment]
Jenkins, test this please.

**Comment by okapies** [comment]
The failure seems to be a precision problem. Going to merge.

**Pull request merged**